ncu: add use cases, use long options (#11555)
tldr-pages/tldr: pages/common/ncu.md

`ncu -i`

> - List minor outdated dependencies:
> 
> `ncu --target minor`

Reviewer: ```suggestion
- List outdated dependencies up to the highest minor version:

`ncu --target {{minor}}`
```

> - List outdated dependencies that match a regex:
> 
> `ncu --filter {{/react/}}`

Reviewer: ```suggestion
- List outdated dependencies that match a keyword or regular expression:

`ncu --filter {{keyword|/regex/}}`
```

- List only outdated dev dependencies

`ncu --dep dev`

- Display help:

> `ncu -h`

Reviewer 2: ```suggestion
`ncu --help`
```
